packages/blade/src/components/Radio/useRadio.ts

```diff
@@ -0,0 +1,158 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import React from 'react';
+import type { GestureResponderEvent } from 'react-native';
+import { useControllableState } from '~src/hooks/useControllable';
+import { useId } from '~src/hooks/useId';
+import { getPlatformType, makeAccessible } from '~src/utils';
+
+export type OnChange = ({
+  isChecked,
+  event,
+  value,
+}: {
+  isChecked: boolean;
+  event?: React.ChangeEvent;
+  value?: string;
+}) => void;
+
+type UseRadioProps = {
+  hasError?: boolean;
+  /**
+   * If `true`, The Radio will be checked. This also makes the Radio controlled
+   * Use `onChange` to update its value
+   *
+   * @default false
+   */
+  isChecked?: boolean;
+  /**
+   * If `true`, the Radio will be initially checked. This also makes the Radio uncontrolled
+   *
+   * @default false
+   */
+  defaultChecked?: boolean;
+  /**
+   * The callback invoked when the checked state of the `Radio` changes.
+   */
+  onChange?: OnChange;
+  /**
+   * The name of the input field in a Radio
+   * (Useful for form submission).
+   */
+  name?: string;
+  /**
+   * The value to be used in the Radio input.
+   * This is the value that will be returned on form submission.
+   */
+  value?: string;
+  /**
+   * If `true`, the Radio will be disabled
+   *
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * If `true`, the Radio input is marked as required,
+   * and `required` attribute will be added
+   *
+   * @default false
+   */
+  isRequired?: boolean;
+};
+
+const useRadio = ({
+  isChecked,
+  defaultChecked,
+  isDisabled,
+  isRequired,
+  hasError,
+  onChange,
+  name,
+  value,
+}: UseRadioProps) => {
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  const isReactNative = getPlatformType() === 'react-native';
+  if (isChecked && defaultChecked) {
+    throw new Error(
+      `[Blade Radio] Do not provide both 'isChecked' and 'defaultChecked' to useRadio. Consider if you want this component to be controlled or uncontrolled.`,
```

Comment:
Will this error be shown to the user if they pass `isChecked` & `defaultChecked` together? If yes then we need to rephrase this. User shouldn't be worried about what `useRadio` is

```suggestion
      `[Blade: Radio] Do not provide both 'isChecked' and 'defaultChecked' to useRadio. Consider if you want this component to be controlled or uncontrolled.`,
```